src/EthereumConnection.js

```diff
@@ -52,29 +56,19 @@ class EthereumConnection extends EventEmitter {
 
         this.accounts = null;
 
-        this.ETHEREUM_CONNECTION_CHECK_INTERVAL =
-            options.ETHEREUM_CONNECTION_CHECK_INTERVAL ||
-            process.env.ETHEREUM_CONNECTION_CHECK_INTERVAL ||
-            DEFAULT_ETHEREUM_CONNECTION_CHECK_INTERVAL;
-
-        this.ETHEREUM_CONNECTION_TIMEOUT =
-            options.ETHEREUM_CONNECTION_TIMEOUT ||
-            process.env.ETHEREUM_CONNECTION_TIMEOUT ||
-            DEFAULT_ETHEREUM_CONNECTION_TIMEOUT;
-
-        this.ETHEREUM_ISLISTENING_TIMEOUT =
-            options.ETHEREUM_ISLISTENING_TIMEOUT ||
-            process.env.ETHEREUM_ISLISTENING_TIMEOUT ||
-            DEFAULT_ETHEREUM_ISLISTENING_TIMEOUT;
-
-        this.ETHEREUM_CONNECTION_CLOSE_TIMEOUT =
-            options.ETHEREUM_CONNECTION_CLOSE_TIMEOUT ||
-            process.env.ETHEREUM_CONNECTION_CLOSE_TIMEOUT ||
-            DEFAULT_ETHEREUM_CONNECTION_CLOSE_TIMEOUT;
-
-        this.PROVIDER_TYPE = options.PROVIDER_TYPE || process.env.PROVIDER_TYPE;
-        this.PROVIDER_URL = options.PROVIDER_URL || process.env.PROVIDER_URL;
-        this.INFURA_PROJECT_ID = options.INFURA_PROJECT_ID || process.env.INFURA_PROJECT_ID || "";
+        const options = Object.assign({}, DEFAULTS, runtimeOptions);
+
+        this.ETHEREUM_CONNECTION_CHECK_INTERVAL = options.ETHEREUM_CONNECTION_CHECK_INTERVAL;
```

Comment:
any point to "copy" these instead of referencing directly via `options.` later? The only reason I can think is if we want to change options on the fly then return to the original options. but even so it migth be better with this.initialOptions and this.currentOptions. So I would rather change direct reference everywhere to this.options and remove these

Reply:
if we just change it to this.options, then no one knows, which properties that object has. I think it's better to have written properties (like this.ETHEREUM_CONNECTION_CHECK_INTERVAL = something) than just referencing on.

Reply:
Typescript ...